### PR TITLE
Improvements to publish github action.

### PR DESCRIPTION
**Pull request by jguiditta**
This patch attempts to any remaining issues from #88, specifically
the mutliple runs (by specifying one one python version to build for)
and not requiring a 'click release button' step to be added to our
process. We may, in a future patch, decide to do more around the
github 'release' concept, which is separate from our current process
of doing a release to pypi when the project is tagged.

This patch also adds a push to the test instance of pypi for quicker
and more frequent verification that pulishing works as expected, and
to give us an easy way for new features to be tested by users between
official tags/releases.

This patch uses the github action built by pypa instead of twine
directly, as we did before, following official docs [1],[2].

[1]
https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/
[2] https://github.com/pypa/gh-action-pypi-publish

Signed-off-by: Jason Guiditta <jguiditt@redhat.com>

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -1,28 +1,36 @@
...
+    - name: Publish to Test PyPI
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        user: __token__
+        password: ${{ secrets.TEST_PYPI_TOKEN }}
+        repository_url: https://test.pypi.org/legacy/
+    - name: Publish to PyPI
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        password: ${{ secrets.TEST_PYPI_TOKEN }}
```
Comment (fuzzball81) [inline .github/workflows/publish.yml:32]: Does this need to be a different token, or is the same used for test and release?

**Review by fuzzball81 — COMMENTED** [review]
Overall this looks good. One minor question.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (jguiditta) [reply .github/workflows/publish.yml:32]: These are different tokens (and I already created them and added them to github)

Comment (fuzzball81) [reply .github/workflows/publish.yml:32]: Let me rephrase. Should we be using TEST_PYPI_TOKEN here or a non-test token?

Comment (jguiditta) [reply .github/workflows/publish.yml:32]: Ah! the snippet here doesn't show enough of the code, so I didn't realize what you meant.  Good catch, this line should be using the non-test one, will update

**Review by fuzzball81 — APPROVED** [review]
LGTM

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -1,28 +1,34 @@
...
+      run:  python -m build
+    - name: Publish to Test PyPI
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        user: __token__
+        password: ${{ secrets.TEST_PYPI_TOKEN }}
+        repository_url: https://test.pypi.org/legacy/
+    - name: Publish to PyPI
+      uses: pypa/gh-action-pypi-publish@master
+      if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
```
Comment (jpichon) [inline .github/workflows/publish.yml:31]: Ooh I was wondering when I saw "push" if we wouldn't be releasing too often. I like this. We're unlikely to tag anything other than releases so this makes sense to me.

**Review by jpichon — APPROVED** [review]

**Comment by jpichon** [comment]
Merging manually since changes to github actions cannot be merged automatically.
